### ISSUE TITLE
2M publishes r_eff_liq = 0 for supercooled liquid, so 66% of liquid cloud radiates at the constant 11 um fallback

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#727]
## What is wrong

`lohmann_2m/scheme.py:668` sets

```python
liquid_cloud_flag = temperature > params.tmelt
```

and passes it as the liquid mask for the effective-radius calculation in
`update_tendencies_and_important_vars`. The radius is masked to exactly 0.0
wherever that flag is False, i.e. in **every cell at or below 273.15 K** —
all supercooled liquid.

`rrtmgp.py` selects on `r_eff_liq_um > 0.0`, so those cells fall back to
`effective_radius_liquid(cdnc_factor, land_fraction)` — the **constant 11.0 um**
described in #717. The 2M path therefore has the same defect #717 documents for
1M, just confined to the mixed-phase regime.

## Measured scope

`echam-rrtmgp-2m` reference run (`jam_runs/rad_emul_ctrl`, T63L47, 2015-01-01, day 1),
all in-cloud liquid points (cloud_fraction > 0.01 and qc > 0), n = 108,957:

| | share |
|---|---|
| supercooled (T <= tmelt) | 66.2% |
| ...of those with `r_eff_liq == 0` | 99.8% |
| warm (T > tmelt) with `r_eff_liq == 0` | 0.2% |
| **all in-cloud liquid on the 11 um fallback** | **66.1%** |

So the prognostic droplet radius the 2M scheme computes is being used for
roughly a third of the liquid cloud; the majority radiates at a constant.

## Why the mask is probably wrong

Supercooled liquid is radiatively liquid — its droplets have a size
distribution and RRTMGP has a liquid LUT for them. The mask that belongs on a
droplet-radius calculation is "is there liquid water here", which the helper
already applies separately (`liquid_in_cloud > 0.0`); the extra
`temperature > tmelt` conjunct removes physically present droplets.

The likely origin is that `ld_liqcl` in ECHAM gates *warm* microphysics
(autoconversion, accretion), where a T > tmelt test is correct, and the same
flag was reused for the radius. That should be confirmed against
`mo_cloud_micro_2m.f90` before changing it — **the ECHAM-HAM source is not
available in this repo** (the in-tree `fortran_harness` carries only
`cumastr`), so this needs the reference, not a judgement call.

## Where the hooks are

- `jcm/physics/clouds/lohmann_2m/scheme.py:668` — the flag.
- `jcm/physics/clouds/lohmann_2m/assembly.py` — passes it to the shared helper.
- `jcm/physics/clouds/cloud_utils.py::eff_liquid_droplet_radius` — takes
  `liquid_cloud_flag` as an optional extra mask; the 1M caller added in #717
  deliberately passes **no** temperature mask, so the two schemes currently
  disagree on this point.

## Blast radius

Changing it is climate-changing for every 2M configuration (`echam-jam`,
`echam-jam-aci`, `echam-rrtmgp-2m`) — it alters shortwave cloud optical depth
throughout the mixed-phase regime. Needs an A/B, not a drive-by.

Found while implementing #717 (shared radius helper) and #725 (ice radius
saturation); not fixed there because both required 2M liquid numerics to stay
bit-identical. Refs #717, #725, #724.

**Comment by duncanwp** [comment]
This appears to have been fixed on `dev` by #707's consolidation, independently of the temperature-split code this issue was filed against: the scan now computes ECHAM's faithful phase-presence flags (`scheme.py`, "8-prep" block) —

```python
# ECHAM ll_liqcl/ll_icecl (1755-1760): actual condensate + number
# above its floor — NOT a temperature split.
ll_liqcl_k = jnp.logical_and(zxlb > params.epsec, cdnc_c >= cdnc_min_k)
ll_icecl_k = jnp.logical_and(zxib > params.epsec, icnc_c >= params.icemin)
```

— and these feed `update_tendencies_and_important_vars`' radius gate, so supercooled liquid with condensate and droplet number now publishes a real `r_eff_liq` instead of 0. The #707 comment cites the ECHAM reference lines (mo_cloud_micro_2m.f90:1755-1760), which is the reference check this issue said was needed.

Suggest verifying the supercooled fraction publishing a nonzero radius on a current-dev run (the 66% figure was measured pre-#707) and then closing.
